conda/core/python_dist.py

```diff
@@ -0,0 +1,929 @@
+            print('ERROR', error)
+
+    return pydist, sp_reference, package_type
+
+
+def get_python_record(anchor_file, prefix_path, context):
+    """
+    Convert a python package defined by an anchor file (Metadata information)
+    into a conda prefix record object.
+
+    Return `None` if the python record cannot be created.
+    """
+    # TODO: ensure that this dist is actually the dist that matches conda-meta
+    pydist, sp_reference, package_type = get_python_distribution_info(anchor_file, prefix_path)
+
+    if pydist is None:
+        return None
+
+    pypi_name = pydist.norm_name
+    conda_name = pypi_name_to_conda_name(pypi_name)
+
+    # TODO: Handle packages with different names (graphviz vs python-graphviz)
+    channel_name = 'pypi' if conda_name == pypi_name else 'pypi:' + pypi_name
+    channel = Channel(channel_name)
+    build = 'pypi_0'
+
+    if package_type == PackageType.SHADOW_PYTHON_EGG_INFO_FILE:
+        paths_data = None
+    elif package_type in (PackageType.SHADOW_PYTHON_DIST_INFO,
+                          PackageType.SHADOW_PYTHON_EGG_INFO_DIR):
+        paths_data = pydist.get_paths_data()
+    elif package_type == PackageType.SHADOW_PYTHON_EGG_LINK:
+        paths_data = pydist.get_paths_data()
+        channel = Channel('<develop>')
+        build = 'dev_0'
+    else:
+        raise NotImplementedError()
+
+    # TODO: This is currently adding additional conda dependencies for graphviz
+    # only, but other packages need something similar. This info should (could)
+    # be on the 'external_requirements', but that field is free form.
+    dependencies = pydist.get_dependencies(context)
+    extra_dependencies = PYPI_CONDA_DEPS.get(pypi_name, [])
+    all_dependencies = list(dependencies) + extra_dependencies
+    # print('{0} ({1})'.format(conda_name, pypi_name))
+    # for dependency in sorted(all_dependencies):
+    #     print('\t{}'.format(dependency))
+    # print('\n')
+
+    python_rec = PrefixRecord(
+        package_type=package_type,
+        name=conda_name,
+        version=pydist.version,
+        channel=channel,
+        subdir='pypi',
+        fn=sp_reference,
+        build=build,
+        build_number=0,
+        paths_data=paths_data,
+        depends=all_dependencies,
+    )
+
+    return python_rec
+
+
+def get_python_records(anchor_files, prefix_path, python_version):
+    """
+    Process all anchor files and return a python record.
+
+    This method evaluates the context needed for marker evaluation.
+    """
+    python_records = []
+    context = update_marker_context(python_version)
+    for anchor_file in sorted(anchor_files):
+        python_record = get_python_record(anchor_file, prefix_path, context)
+        if python_record:
+            python_records.append(python_record)
+    return python_records
+
+
+# Marker helper funcs
+# -----------------------------------------------------------------------------
+def evaluate_marker(marker_expr, context, extras):
```

Comment:
@kalefranz I wanted to do a POC for marker evaluation and wrote this.

For the tests packages we get something like (pretty printed)

```
anaconda-client (anaconda-client)
	python ==3.6


conda (conda)
	python ==3.6


python-graphviz (graphviz)
	flake8
	graphviz
	mock >=2
	pep8-naming
	pytest >=3.4
	pytest-cov
	pytest-mock >=1.8
	python >=2.7,!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*
	sphinx >=1.3
	sphinx-rtd-theme
	tox >=3.0
	twine
	wheel


libsass (libsass)
	python ==3.6
	six
	twine ==1.5.0


loghub (loghub)
	jinja2
	python ==3.6
```

And

```
babel (babel)
	python >=2.7,!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*
	pytz >=0a


django (django)
	argon2-cffi >=16.1.0
	bcrypt
	python ==2.7
	pytz


pyjwt (pyjwt)
	cryptography >=1.4
	flake8
	flake8-import-order
	pep8-naming
	pytest <4,>3
	pytest-cov
	pytest-runner
	python ==2.7


asn1crypto (asn1crypto)
	python ==2.7


cffi (cffi)
	pycparser
	python ==2.7


chardet (chardet)
	python ==2.7


cryptography (cryptography)
	asn1crypto >=0.21.0
	cffi !=1.11.3,>=1.7
	cryptography-vectors ==2.3
	doc8
	enum34
	flake8
	flake8-import-order
	hypothesis >=1.11.4
	idna >=2.1
	ipaddress
	iso8601
	pep8-naming
	pretend
	pyenchant >=1.6.11
	pytest !=3.3.0,>=3.2.1
	python >=2.7,!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*
	pytz
	readme-renderer >=16.0
	six >=1.4.1
	sphinx >=1.6.5
	sphinx-rtd-theme
	sphinxcontrib-spelling >=4.0.1


dask (dask)
	cloudpickle >=0.2.1
	distributed >=1.22
	numpy >=1.11.0
	pandas >=0.19.0
	partd >=0.3.8
	python ==2.7
	toolz >=0.7.3


django-phonenumber-field (django-phonenumber-field)
	babel
	django >=1.11
	phonenumberslite >=7.0.2
	python ==2.7


django-twilio (django-twilio)
	django <2,>=1.8
	django-phonenumber-field >=0.6
	python ==2.7
	setuptools >=36.2
	twilio <7,>=6.3.0


enum34 (enum34)
	python ==2.7


h5py (h5py)
	numpy >=1.7
	python ==2.7
	six


hdf5storage (hdf5storage)
	h5py >=2.1
	numpy
	python ==2.7


idna (idna)
	python ==2.7


ipaddress (ipaddress)
	python ==2.7


numpy (numpy)
	python >=2.7,!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*


packaging (packaging)
	pyparsing >=2.0.2
	python >=2.7,!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*
	six


phonenumberslite (phonenumberslite)
	python ==2.7


pluggy (pluggy)
	python >=2.7,!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*


py (py)
	python >=2.7,!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*


pyopenssl (pyopenssl)
	cryptography >=2.2.1
	flaky
	pretend
	pytest >=3.0.1
	python ==2.7
	six >=1.5.2
	sphinx
	sphinx-rtd-theme


pycparser (pycparser)
	python ==2.7


pyparsing (pyparsing)
	python ==2.7


pytz (pytz)
	python ==2.7


requests (requests)
	certifi >=2017.4.17
	chardet <3.1.0,>=3.0.2
	cryptography >=1.3.4
	idna <2.8,>=2.5
	idna >=2.0.0
	pyopenssl >=0.14
	pysocks !=1.5.7,>=1.5.6
	python >=2.6,!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*
	urllib3 <1.24,>=1.21.1


six (six)
	python ==2.7


tox (tox)
	packaging >=17.1
	pluggy <1,>=0.3.0
	py <2,>=1.4.17
	pytest <4,>=3.0.0
	pytest-cov <3,>=2.5.1
	pytest-mock <2,>=1.10.0
	pytest-randomly <2,>=1.2.3
	pytest-timeout <2,>=1.3.0
	pytest-xdist <2,>=1.22.2
	python >=2.7,!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*
	six <2,>=1.0.0
	sphinx <2,>=1.7.5
	towncrier >=18.5.0
	virtualenv >=1.11.2


twilio (twilio)
	pyjwt >=1.4.2
	python ==2.7
	pytz
	requests >=2.0.0
	six


urllib3 (urllib3)
	certifi
	cryptography >=1.3.4
	idna >=2.0.0
	ipaddress
	pyopenssl >=0.14
	pysocks >=1.5.6,<2.0,!=1.5.7
	python >=2.6,!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*,<4


virtualenv (virtualenv)
	python >=2.7,!=3.0.*,!=3.1.*,!=3.2.*
```